includes/Checker/Abstract_Check_Runner.php

```diff
@@ -243,16 +251,46 @@ private function get_shared_preparations( array $checks ) {
 	 * @since n.e.x.t
 	 *
 	 * @return array An array map of check slugs to Check instances.
+	 *
+	 * @throws Exception Thrown exception when a runtime check is requested and the plugin inactive.
 	 */
-	protected function get_checks_to_run() {
-		$check_slugs = $this->get_check_slugs();
-		$all_checks  = $this->get_checks_instance()->get_checks();
+	public function get_checks_to_run() {
+		$check_slugs   = $this->get_check_slugs();
+		$all_checks    = $this->get_checks_instance()->get_checks();
+		$plugin_active = is_plugin_active( $this->get_plugin_basename() );
+
+		if ( ! empty( $check_slugs ) ) {
+			// Get the check instances based on the requested checks.
+			$checks_to_run = array_intersect_key( $all_checks, array_flip( $check_slugs ) );
+
+			// Check the following conditions if at least one runtime check is requested.
+			if ( $this->has_runtime_check( $checks_to_run ) ) {
+				// Throw an error if the plugin is not active.
+				if ( ! $plugin_active ) {
+					throw new Exception( __( 'Runtime checks cannot be run against inactive plugins.', 'plugin-check' ) );
+				}
 
-		if ( empty( $check_slugs ) ) {
-			return $all_checks;
+				// Throw and error if the runner was not initialized early and the runtime environment was not set up.
+				if ( ! $this->initialized_early ) {
+					throw new Exception( __( 'Runtime checks cannot be run as the runtime environment was not set up.', 'plugin-check' ) );
+				}
+			}
+		} else {
+			// Run all checks for the plugin.
+			$checks_to_run = $all_checks;
+
+			// Only run static checks if the plugin is inactive or the runtime environment was not set up.
+			if ( ! $plugin_active || ! $this->initialized_early ) {
+				$checks_to_run = array_filter(
+					$checks_to_run,
+					function ( $check ) {
+						return ! $check instanceof Runtime_Check;
```

Comment:
This line is repeated
https://github.com/10up/plugin-check/pull/100/files#diff-ad151f5d51d23b5a7893d014606fa6d982f5efce1a005d4070a44957badfb35eR207

A little util method could be used here.